app/components/primer/heading_component.rb

```diff
@@ -1,19 +1,47 @@
 # frozen_string_literal: true
 
 module Primer
-  # Use the Heading component to wrap a component that will create a heading element
+  # Heading can be used to communicate page organization and hierarchy.
+  #
+  # - Set tag to one of `:h1`, `:h2`, `:h3`, `:h4`, `:h5`, `:h6` based on what is
+  #   appropriate for the page context.
+  # - Use Heading as the title of a section or sub section.
+  # - Do not use Heading for styling alone. To style text without conveying heading semantics,
+  #   consider using <%= link_to_component(Primer::TextComponent) %> with relevant <%= link_to_typography_docs %>.
+  # - Do not jump heading levels. For instance, do not follow a `<h1>` with an `<h3>`. Heading levels should
+  #   increase by one in ascending order.
+  #
+  # @accessibility
+  #   Headings convey semantic meaning. Assistive technology users rely on headings to quickly navigate and scan information on a page.
+  #   Inappropriate use of heading can lead to a confusing experience so be sure to follow best practices.
+  #   [Learn more about best heading practices (WAI Headings)](https://www.w3.org/WAI/tutorials/page-structure/headings/)
   class HeadingComponent < Primer::Component
     status :beta
 
+    TAG_OPTIONS = [:h1, :h2, :h3, :h4, :h5, :h6].freeze
+    TAG_FALLBACK = :h2
+
```

Comment:
If you want to be fancy:

```suggestion
    TAG_OPTIONS = [:h1, TAG_FALLBACK, :h3, :h4, :h5, :h6].freeze
    TAG_FALLBACK = :h2

```

Reply:
i love ✨fancy